Implemented Single Sign On using Azure

nmcmahon-unosquare opened the pull request:
Added logout functionality to AuthenticationController
Added SingleSignOnController to deal specifically with Azure AD logins.
Enabled SSL for project (necessary for Azure AD logins)
Installed necessary NuGet Packages.
Added Azure config to Startup.cs
Removed Authenticated user and authentication service. These are not needed anymore.
Added ability to retrieve logged in user details from Azure in controllers.

Jamie-Unosquare commented on v2/AdminCore.Services.Tests/EventServiceTests.cs:
```diff
@@ -45,7 +42,7 @@ public void CreateEvent_WhenCalled_SuccessfullyInsertsNewEventIntoDb()
 
       // Act
-      _eventService.CreateEvent(eventDateDto, EventTypes.AnnualLeave);
+      _eventService.CreateEvent(eventDateDto, EventTypes.AnnualLeave, 1);
```
Create a local variable for this employeeId

nmcmahon-unosquare replied:
Fixed

Jamie-Unosquare commented on v2/AdminCore.Services/EventService.cs:
```diff
@@ -366,14 +363,14 @@ private EventDto ValidateRemainingHolidaysAndCreate(Event newEvent)
     }
 
-    private Event BuildNewEvent(int employeeId, EventTypes eventTypes)
+    private Event BuildNewEvent(int employeeId)
```
This overrides a previous fix, revert this

nmcmahon-unosquare replied:
Fixed

Jamie-Unosquare commented on v2/AdminCore.Services/EventService.cs:
```diff
@@ -366,14 +363,14 @@ private EventDto ValidateRemainingHolidaysAndCreate(Event newEvent)
         EmployeeId = employeeId,
         EventStatusId = (int)EventStatuses.AwaitingApproval,
-        EventTypeId = (int)eventTypes,
+        EventTypeId = (int)EventTypes.AnnualLeave,
```
See above

nmcmahon-unosquare replied:
Fixed

Jamie-Unosquare commented on v2/AdminCore.WebApi.Tests/Controllers/HolidayControllerTests.cs:
```diff
@@ -46,15 +43,14 @@ public void CreateHoliday_WhenCalled_ReturnsCreatedHoliday()
 
       VerifyActionResult(result);
-      _eventService.Received(1).CreateEvent(Arg.Any<EventDateDto>(), EventTypes.AnnualLeave);
+      _eventService.Received(1).CreateEvent(Arg.Any<EventDateDto>(), EventTypes.AnnualLeave, 1);
```
See above, local variable for this

nmcmahon-unosquare replied:
Fixed

Jamie-Unosquare reviewed (changes requested):
Biggest issue is reverting the eventtypes

nmcmahon-unosquare committed:
Implemented Single Sign On using Azure

Added logout functionality to AuthenticationController
Added SingleSignOnController to deal specifically with Azure AD logins.
Enabled SSL for project (necessary for Azure AD logins)
Installed necessary NuGet Packages.
Added Azure config to Startup.cs
Removed Authenticated user and authentication service. These are not needed anymore.
Added ability to retrieve logged in user details from Azure in controllers.

BrownMark commented on v2/AdminCore.Services/AuthenticatedUser.cs:
```diff
@@ -1,22 +0,0 @@
-
-    public int RetrieveUserId()
-    {
-      var employeeId = _httpContentAccessor.HttpContext.User.FindFirst(ClaimTypes.Name).Value;
```
Can you give me a shout about this 👍 . Just want to check something
